plugins/hls-splice-plugin/src/Ide/Plugin/Splice.hs

```diff
@@ -0,0 +1,488 @@
+            , ""
+            , T.pack (show warns)
+            ]
+    pure resl
+    where
+        dflags = hsc_dflags hscEnv
+
+-- | FIXME:  Is thereAny "clever" way to do this exploiting TTG?
+unRenamedE ::
+    forall ast m.
+    (Fail.MonadFail m, HasSplice ast) =>
+    DynFlags ->
+    ast GhcRn ->
+    TransformT m (Located (ast GhcPs))
+unRenamedE dflags expr = do
+    uniq <- show <$> uniqueSrcSpanT
+    (anns, expr') <-
+        either (fail . show) pure $
+            parseAST @(ast GhcPs) dflags uniq $
+                showSDoc dflags $ ppr expr
+    let _anns' = setPrecedingLines expr' 0 1 anns
+    pure expr'
+
+data SearchResult r =
+    Continue | Stop | Here r
+    deriving (Read, Show, Eq, Ord, Data, Typeable)
+
+fromSearchResult :: SearchResult a -> Maybe a
+fromSearchResult (Here r) = Just r
+fromSearchResult _ = Nothing
+
+-- TODO: workaround when HieAst unavailable (e.g. when the module itself errors)
+-- TODO: Declaration Splices won't appear in HieAst; perhaps we must just use Parsed/Renamed ASTs?
+codeAction :: CodeActionProvider IdeState
+codeAction _ state plId docId ran _ =
+    fmap (maybe (Right $ List []) Right) $
+        runMaybeT $ do
+            fp <- MaybeT $ pure $ uriToNormalizedFilePath $ toNormalizedUri theUri
+            ParsedModule {..} <-
+                MaybeT . runAction "splice.codeAction.GitHieAst" state $
+                    use GetParsedModule fp
+            let spn =
+                    rangeToRealSrcSpan ran $
+                        fromString $
+                            fromNormalizedFilePath fp
+                mouterSplice = something' (detectSplice spn) pm_parsed_source
+            mcmds <- forM mouterSplice $
+                \(spliceSpan, spliceContext) ->
+                    forM expandStyles $ \(_, (title, cmdId)) -> do
+                        let params = ExpandSpliceParams {uri = theUri, ..}
+                        act <- liftIO $ mkLspCommand plId cmdId title (Just [toJSON params])
+                        pure $
+                            CACodeAction $
+                                CodeAction title (Just CodeActionRefactorRewrite) Nothing Nothing (Just act)
+
+            pure $ maybe mempty List mcmds
+    where
+        theUri = docId ^. J.uri
+        detectSplice ::
+            RealSrcSpan ->
+            GenericQ (SearchResult (RealSrcSpan, SpliceContext))
+        detectSplice spn =
+            mkQ
+                Continue
+                ( \case
+                    (L l@(RealSrcSpan spLoc) expr :: LHsExpr GhcPs)
+                        | RealSrcSpan spn `isSubspanOf` l ->
+                            case expr of
+                                HsSpliceE {} -> Here (spLoc, Expr)
+                                _ -> Continue
+                    _ -> Stop
+                )
+#if __GLASGOW_HASKELL__ == 808
+#else
+                `extQ` \case
+                    (L l@(RealSrcSpan spLoc) pat :: LPat GhcPs)
```

Comment:
You can use `(dL -> L l@(RealSrcSpan spLoc) pat :: Located (Pat GhcPs))` for 8.8

https://hackage.haskell.org/package/ghc-8.8.3/docs/GHC.html#v:dL

Reply:
Oh, I wasn't aware of that! Last time, I tried to extract location info from `XPat`, constructor, which gave a slightly different SrcLoc than expected. Contrary, at least on my laptop, it seems that the combination with `cL` and `dL` gives a  correct SrcSpan and Pattern Splice Expansion successes also in GHC 8.8!